templates/an-cash-dao/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,359 @@
+        emit DeployDao(address(dao));
+        acl = ACL(dao.acl());
+        _createPermissionForTemplate(acl, dao, dao.APP_MANAGER_ROLE());
+    }
+
+    /* ACL */
+
+    function _createPermissions(ACL _acl, address[] memory _grantees, address _app, bytes32 _permission, address _manager) internal {
+        _acl.createPermission(_grantees[0], _app, _permission, address(this));
+        for (uint256 i = 1; i < _grantees.length; i++) {
+            _acl.grantPermission(_grantees[i], _app, _permission);
+        }
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.setPermissionManager(_manager, _app, _permission);
+    }
+
+    function _createPermissionForTemplate(ACL _acl, address _app, bytes32 _permission) internal {
+        _acl.createPermission(address(this), _app, _permission, address(this));
+    }
+
+    function _removePermissionFromTemplate(ACL _acl, address _app, bytes32 _permission) internal {
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.removePermissionManager(_app, _permission);
+    }
+
+    function _transferRootPermissionsFromTemplateAndFinalizeDAO(Kernel _dao, address _to) internal {
+        _transferRootPermissionsFromTemplateAndFinalizeDAO(_dao, _to, _to);
+    }
+
+    function _transferRootPermissionsFromTemplateAndFinalizeDAO(Kernel _dao, address _to, address _manager) internal {
+        ACL _acl = ACL(_dao.acl());
+        _transferPermissionFromTemplate(_acl, _dao, _to, _dao.APP_MANAGER_ROLE(), _manager);
+        _transferPermissionFromTemplate(_acl, _acl, _to, _acl.CREATE_PERMISSIONS_ROLE(), _manager);
+        emit SetupDao(_dao);
+    }
+
+    function _transferPermissionFromTemplate(ACL _acl, address _app, address _to, bytes32 _permission, address _manager) internal {
+        _acl.grantPermission(_to, _app, _permission);
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.setPermissionManager(_manager, _app, _permission);
+    }
+
+    /* AGREEMENT */
+
+    function _installAgreementApp(
+        Kernel _dao,
+        address _arbitrator,
+        bool _setAppFeesCashier,
+        string _title,
+        bytes _content,
+        address _stakingFactory
+    )
+        internal
+        returns (Agreement)
+    {
+        bytes memory initializeData = abi.encodeWithSelector(Agreement(0).initialize.selector, _arbitrator, _setAppFeesCashier, _title, _content, _stakingFactory);
+        return Agreement(_installNonDefaultApp(_dao, AGREEMENT_APP_ID, initializeData));
+    }
+
+    function _createAgreementPermissions(ACL _acl, Agreement _agreement, address _grantee, address _manager) internal {
+        _acl.createPermission(_grantee, _agreement, _agreement.CHANGE_AGREEMENT_ROLE(), _manager);
+        _acl.createPermission(address(this), _agreement, _agreement.MANAGE_DISPUTABLE_ROLE(), address(this));
+    }
+
+    /* AGENT */
+
+    function _installDefaultAgentApp(Kernel _dao) internal returns (Agent) {
```

Comment:
Similarly, we can probably simplify the Agent/Vault functions since we know we're only going to be installing an Agent?